lib/auth/auth.go

```diff
@@ -3347,6 +3352,71 @@ func (a *Server) DeleteNetworkRestrictions(ctx context.Context) error {
 	return a.Services.Restrictions.DeleteNetworkRestrictions(ctx)
 }
 
+// GetPluginData loads all plugin data matching the supplied filter.
+func (a *Server) GetPluginData(ctx context.Context, filter types.PluginDataFilter) ([]types.PluginData, error) {
+	if !isAllowedPluginDataKind(filter.Kind) {
+		return nil, trace.BadParameter("unsupported resource kind %q", filter.Kind)
+	}
+
+	return a.Services.PluginData.GetPluginData(ctx, filter)
+}
+
+// GetPluginData loads all plugin data matching the supplied filter.
+func (a *Server) UpdatePluginData(ctx context.Context, params types.PluginDataUpdateParams) error {
+	if !isAllowedPluginDataKind(params.Kind) {
+		return trace.BadParameter("unsupported resource kind %q", params.Kind)
+	}
+
+	// Load parent resource expiration value. Nil means that a parent resource does not exist.
+	// Non-existent parent resource is fine on update. There are the cases when we need to access
+	// PluginData of a just deleted resource. We need to prevent creation of PluginData entries
+	// for a missing resources.
+	//
+	// Expires itself is needed to prevent orphaned plugin data, we automatically
+	// configure new instances to expire shortly after the resource to which they are associated.
+	expires, err := a.getResourceExpiry(ctx, params.Resource, params.Kind)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return a.PluginData.UpsertPluginData(ctx, params, expires)
+}
+
+// isAllowedPluginDataKind returns true if PluginData can be attached to a resource kind.
+func isAllowedPluginDataKind(kind string) bool {
+	return kind == types.KindAccessRequest || kind == types.KindUser
+}
```

Comment:
Let's keep the plugin data kind check on the backend, see comment below.
```suggestion
func (a *Server) GetPluginData(ctx context.Context, filter types.PluginDataFilter) ([]types.PluginData, error) {
	return a.Services.PluginData.GetPluginData(ctx, filter)
}

// GetPluginData loads all plugin data matching the supplied filter.
func (a *Server) UpdatePluginData(ctx context.Context, params types.PluginDataUpdateParams) error {
	// Load parent resource expiration value. Nil means that a parent resource does not exist.
	// Non-existent parent resource is fine on update. There are the cases when we need to access
	// PluginData of a just deleted resource. We need to prevent creation of PluginData entries
	// for a missing resources.
	//
	// Expires itself is needed to prevent orphaned plugin data, we automatically
	// configure new instances to expire shortly after the resource to which they are associated.
	expires, err := a.getResourceExpiry(ctx, params.Resource, params.Kind)
	if err != nil {
		return trace.Wrap(err)
	}

	return a.PluginData.UpsertPluginData(ctx, params, expires)
}
```